sources/api/schnauzer/src/helpers.rs

```diff
@@ -1393,38 +1393,34 @@ pub fn oci_defaults(
 ) -> Result<(), RenderError> {
     // To give context to our errors, get the template name (e.g. what file we are rendering), if available.
     debug!("Starting oci_defaults helper");
-    let template_name = template_name(renderctx);
-    debug!("Template name: {}", &template_name);
-
-    // Check number of parameters, must be exactly two (OCI spec section to render and settings values for the section)
-    debug!("Number of params: {}", helper.params().len());
-    check_param_count(helper, template_name, 1)?;
-    debug!("params: {:?}", helper.params());
 
-    debug!("Getting the requested OCI spec section to render");
-    let oci_defaults_values = get_param(helper, 0)?;
-    // We want the settings path so we know which OCI spec section we are rendering.
-    // e.g. settings.oci-defaults.resource-limits
-    let settings_path = get_param_key_name(helper, 0)?;
-    // Extract the last part of the settings path, which is the OCI spec section we want to render.
-    let oci_spec_section = settings_path
-        .split('.')
-        .last()
-        .expect("did not find (got None for) an oci_spec_section");
+    render_oci_defaults(
```

Comment:
Instead of having a new helper, what if you pass another argument to the `oci_defaults` helper, that is used to determine the “format” of the settings?

```handlebars
{{~oci_defaults <settings> docker~}}
```

By default, `oci_defaults` will render the settings for `containerd`. Then in the code that generates the `result_lines` you can have a trait that is implemented for both Docker and containerd:

```rust
# Somewhere define something similar to this trait
trait OciDefaultsFormatter {
   fn get_capabilities(caps: String);
   fn get_resource_limits(limits: String);
}
```

You can pass an instance of a struct that implements this trait to `oci_spec_capabilities` and `oci_spec_resource_limits`, and call either method as needed.